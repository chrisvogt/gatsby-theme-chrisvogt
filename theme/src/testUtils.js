import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { ThemeProvider } from 'theme-ui'
import theme from './gatsby-plugin-theme-ui'

import store from './store'

export { store }

// React module that provides necessary context to components being tested.
export const TestProvider = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const TestProviderWithState = ({ children }) => (
  <ReduxProvider store={store}>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </ReduxProvider>
)
