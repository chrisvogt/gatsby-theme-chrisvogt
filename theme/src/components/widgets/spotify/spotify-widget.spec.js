import React from 'react'
import SpotifyWidget from './spotify-widget'
import renderer from 'react-test-renderer'
import { TestProviderWithState } from '../../../testUtils'
import useSiteMetadata from '../../../hooks/use-site-metadata'

jest.mock('../../../hooks/use-site-metadata')

const mockSiteMetadata = {
  widgets: {
    spotify: {
      username: 'mockusername',
      widgetDataSource: 'https://fake-api.example.com/social/spotify'
    }
  }
}

describe('Spotify Widget', () => {
  beforeEach(() => {
    useSiteMetadata.mockImplementation(() => mockSiteMetadata)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('matches the loading state snapshot', () => {
    const tree = renderer
      .create(
        <TestProviderWithState>
          <SpotifyWidget />
        </TestProviderWithState>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches the loaded state snapshot', () => {
    const initialState = {
      widgets: {
        spotify: {
          state: 'SUCCESS',
          data: {
            collections: {
              playlists: [
                {
                  id: 'playlist1',
                  name: 'Test Playlist',
                  description: 'Test description',
                  images: [{ url: 'https://example.com/image.jpg' }],
                  external_urls: { spotify: 'https://spotify.com/playlist1' }
                }
              ],
              topTracks: [
                {
                  id: 'track1',
                  name: 'Test Track',
                  artists: [{ name: 'Test Artist' }],
                  album: { name: 'Test Album' },
                  external_urls: { spotify: 'https://spotify.com/track1' }
                }
              ]
            },
            metrics: {
              Playlists: 10,
              'Top Tracks': 20
            },
            profile: {
              displayName: 'Test User',
              profileURL: 'https://spotify.com/user/test'
            },
            provider: {
              displayName: 'Spotify'
            }
          }
        }
      }
    }

    const tree = renderer
      .create(
        <TestProviderWithState initialState={initialState}>
          <SpotifyWidget />
        </TestProviderWithState>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches the error state snapshot', () => {
    const initialState = {
      widgets: {
        spotify: {
          state: 'FAILURE',
          data: null
        }
      }
    }

    const tree = renderer
      .create(
        <TestProviderWithState initialState={initialState}>
          <SpotifyWidget />
        </TestProviderWithState>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
