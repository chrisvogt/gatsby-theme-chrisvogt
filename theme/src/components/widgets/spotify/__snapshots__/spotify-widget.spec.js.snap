// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spotify Widget matches the loading state snapshot 1`] = `
<section
  className="css-12h1dx4-Widget"
  id="spotify"
>
  <header
    className="css-9v09o4-WidgetHeader"
  >
    <h2
      className="css-1r48a8i-Box"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-spotify css-4hvi9p-WidgetHeader"
        data-icon="spotify"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={
          {
            "height": "22px",
          }
        }
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111.1 8 0 119.1 0 256s111.1 248 248 248 248-111.1 248-248S384.9 8 248 8zm100.7 364.9c-4.2 0-6.8-1.3-10.7-3.6-62.4-37.6-135-39.2-206.7-24.5-3.9 1-9 2.6-11.9 2.6-9.7 0-15.8-7.7-15.8-15.8 0-10.3 6.1-15.2 13.6-16.8 81.9-18.1 165.6-16.5 237 26.2 6.1 3.9 9.7 7.4 9.7 16.5s-7.1 15.4-15.2 15.4zm26.9-65.6c-5.2 0-8.7-2.3-12.3-4.2-62.5-37-155.7-51.9-238.6-29.4-4.8 1.3-7.4 2.6-11.9 2.6-10.7 0-19.4-8.7-19.4-19.4s5.2-17.8 15.5-20.7c27.8-7.8 56.2-13.6 97.8-13.6 64.9 0 127.6 16.1 177 45.5 8.1 4.8 11.3 11 11.3 19.7-.1 10.8-8.5 19.5-19.4 19.5zm31-76.2c-5.2 0-8.4-1.3-12.9-3.9-71.2-42.5-198.5-52.7-280.9-29.7-3.6 1-8.1 2.6-12.9 2.6-13.2 0-23.3-10.3-23.3-23.6 0-13.6 8.4-21.3 17.4-23.9 35.2-10.3 74.6-15.2 117.5-15.2 73 0 149.5 15.2 205.4 47.8 7.8 4.5 12.9 10.7 12.9 22.6 0 13.6-11 23.3-23.2 23.3z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Spotify
    </h2>
    <div
      className="css-1i3svai-WidgetHeader"
    >
      <svg
        className="svg-loaders-svg css-zmfjn6-CallToAction"
        fill="#1E90FF"
        height="24"
        viewBox="0 0 135 140"
        width="24"
      >
        <rect
          height={120}
          rx={6}
          width={15}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={30}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={140}
          rx={6}
          width={15}
          x={60}
        >
          <animate
            attributeName="height"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={90}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={120}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
      </svg>
    </div>
  </header>
  <div
    className="css-o0oe1r-ProfileMetricsBadge"
  >
    <div
      className="css-1fj08pc-Box"
    >
       
    </div>
    <div
      className="css-1wc3ku-Box"
    >
       
    </div>
  </div>
  <div
    className="css-swnv5q-Playlists"
  >
    <div
      className="css-4i4294-Playlists"
    >
      <h3
        className="css-1j0cp9w-Playlists"
      >
        Playlists
      </h3>
    </div>
    <p
      className="css-1qwovgy"
    >
      My 12 favorite playlists.
    </p>
    <div
      className="media-item_grid null css-1fvffyl-MediaItemGrid"
    >
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="css-kohdhs-TopTracks"
    >
      <h3
        className="css-zhe7fa-TopTracks"
      >
        Top Tracks
      </h3>
    </div>
    <p
      className="css-1qwovgy"
    >
      My 12 most-played tracks over the last 4 weeks.
    </p>
    <div
      className="media-item_grid null css-1fvffyl-MediaItemGrid"
    >
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="show-loading-animation"
      >
        <div
          className="rect-shape css-1fetlaf"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</section>
`;
